spec/facter/facter_spec.rb

```diff
@@ -368,7 +368,7 @@ def mock_collection(method, os_name = nil, error = nil)
       it 'sets show_legacy to true' do
         Facter.values({}, [])
 
-        expect(Facter::Options[:show_legacy]).to be true
+        expect(Facter::Options[:show_legacy]).to be false
```

Comment:
nit, the description still says `sets show_legacy to true`.

It might be good to add a test that shows how the behavior changes when `show_legacy` is `true` vs `false`. For example, when true `Facter.values` returns an `uptime` fact. When `false` it doesn't.

Reply:
Good catch! I've added integration tests to show the behaviour in the two cases you described.